src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlAuthenticationProviderManager.NetCoreApp.cs

```diff
@@ -0,0 +1,114 @@
+
+namespace Microsoft.Data.SqlClient
+{
+    internal partial class SqlAuthenticationProviderManager
+    {
+        private readonly SqlAuthenticationInitializer _initializer;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        public SqlAuthenticationProviderManager(SqlAuthenticationProviderConfigurationSection configSection = null)
+        {
+            var methodName = "Ctor";
+            _typeName = GetType().Name;
+            _providers = new ConcurrentDictionary<SqlAuthenticationMethod, SqlAuthenticationProvider>();
+            var authenticationsWithAppSpecifiedProvider = new HashSet<SqlAuthenticationMethod>();
+            _authenticationsWithAppSpecifiedProvider = authenticationsWithAppSpecifiedProvider;
+
+            if (configSection == null)
+            {
+                _sqlAuthLogger.LogInfo(_typeName, methodName, "No SqlAuthProviders configuration section found.");
+                return;
+            }
+
+            // Create user-defined auth initializer, if any.
+            if (!string.IsNullOrEmpty(configSection.InitializerType))
+            {
+                try
+                {
+                    var initializerType = Type.GetType(configSection.InitializerType, true);
+                    _initializer = (SqlAuthenticationInitializer)Activator.CreateInstance(initializerType);
+                    _initializer.Initialize();
+                }
+                catch (Exception e)
+                {
+                    throw SQL.CannotCreateSqlAuthInitializer(configSection.InitializerType, e);
+                }
+                _sqlAuthLogger.LogInfo(_typeName, methodName, "Created user-defined SqlAuthenticationInitializer.");
+            }
+            else
+            {
+                _sqlAuthLogger.LogInfo(_typeName, methodName, "No user-defined SqlAuthenticationInitializer found.");
+            }
+
+            // add user-defined providers, if any.
+            if (configSection.Providers != null && configSection.Providers.Count > 0)
+            {
+                foreach (ProviderSettings providerSettings in configSection.Providers)
+                {
+                    SqlAuthenticationMethod authentication = AuthenticationEnumFromString(providerSettings.Name);
+                    SqlAuthenticationProvider provider;
+                    try
+                    {
+                        var providerType = Type.GetType(providerSettings.Type, true);
+                        provider = (SqlAuthenticationProvider)Activator.CreateInstance(providerType);
+                    }
+                    catch (Exception e)
+                    {
+                        throw SQL.CannotCreateAuthProvider(authentication.ToString(), providerSettings.Type, e);
+                    }
+                    if (!provider.IsSupported(authentication))
+                        throw SQL.UnsupportedAuthenticationByProvider(authentication.ToString(), providerSettings.Type);
```

Comment:
Maybe add curly braces for readability?

Reply:
Sure, I just picked it raw from another class.